sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp

```diff
@@ -182,6 +183,40 @@ namespace Azure {
         KeyWrapAlgorithm algorithm,
         std::vector<uint8_t> const& encryptedKey,
         Azure::Core::Context const& context = Azure::Core::Context());
+
+    /**
+     * @brief Signs the specified digest.
+     *
+     * @param algorithm The #SignatureAlgorithm to use.
+     * @param digest The pre-hashed digest to sign. The hash algorithm used to compute the digest
+     * must be compatable with the specified algorithm.
+     * @param context A #Azure::Core::Context to cancel the operation.
+     * @return The result of the sign operation. The returned #SignResult contains the signature
+     * along with all other information needed to verify it. This information should be stored with
+     * the signature.
+     */
+    SignResult Sign(
```

Comment:
Were you not going to add the SignData and VerifyData methods? Effectively, these do the hashing for you as well. The methods as defined here require the caller to hash data and produce the digest themselves. Or is that coming in a subsequent PR?

On that note, could you get someone to review the cpp - or I can try to find some time to later - and merge these, making subsequent PRs?

Reply:
>On that note, could you get someone to review the cpp - or I can try to find some time to later - and merge these, making subsequent PRs?

Yes, I am waiting also on @RickWinter  @ahsonkhan  and @antkmsft for review the impl,  However, most of the specific implementation like the local algorithms are not yet added.

> Were you not going to add the SignData and VerifyData methods?

I will. But I we only have MD5 hash support for now on Azure Core. So, I need to add SHA256/384/512 before adding the SingData/VerifyData.   Since Azure-Core is close to GA, and we are trying not to add more features close to GA, I will add this as part of KeyVault Common for now and move it to Core in the future.

Reply:
MD5 hashing? In 2021?! Even SHA1 has been taboo for a few years.

Reply:
I have added Sing/Verify Data for Linux  (using openssl for generating SHA256/384/512)

I will update next for Windows

Reply:
> MD5 hashing? In 2021?! Even SHA1 has been taboo for a few years.

There are a couple storage APIs that only support MD5 :(